### PR TITLE
Scrap Trader Inventory Expansion and QoL Update

### PR DESCRIPTION
**Pull request by Merikona**
#### Summary
Features "Scrap trader now sells bronze and aluminum and is no longer limited by pocket size when trading"

#### Purpose of change
I made an issue a couple days ago, here https://github.com/CleverRaven/Cataclysm-DDA/issues/59749, because I realized that it made little sense that someone who is sedentary is restricted by their pocket size when trading, and furthermore, that there should be plenty of metals beyond steel that he should have access to in a scrap yard. 

Edit: I didn't realize at the time of making this PR that *spoiler ahead* some specialty metals are locked behind a quest-chain. This means I'll have to revise a bit of my intended changes. I still want to add bronze and aluminum though. 

#### Describe the solution
Allows scrap trader to accept larger volume items and gives 2 additional dialogue options for buying aluminum and bronze when trading. I consulted some real-world pricing for various metals here, https://rockawayrecycling.com/scrap-metal-prices/, but ultimately determined that the pricing would change for a post-apocalyptic scenario. e.g. bronze would be higher-valued than copper because it's more durable and can be reforged into armor. 

Edit: Also changes to the mapgen, including a tow truck and shed. 

#### Describe alternatives you've considered
Considered adding wood to the scrap trader's inventory as well since it made sense to me he'd have plenty of access to scrap wood, but figured I'll wait on that bit and maybe ask for feedback first. 

#### Testing
Created a world and character in my branched version of CDDA. Teleported to a scrap trader. Traded him a log in exchange for his goods to test his volume limits. Used the three new dialogue options to obtain scrap aluminum, scrap copper, and bronze.

### CODE REVIEW

**Comment by Merikona** [comment]
Ran into a curious problem testing. Items you trade to scrapper do not disappear over time. I've found a solution but have yet to implement it. Going to rework this NPC to have a weekly stock as well when I do. So marked as draft for now. To be continued.

**Comment by MNG-cataclysm** [comment]
I don't think the scrap trader should sell wood; the lumber merchants already do that.

**Comment by Merikona** [comment]
> I don't think the scrap trader should sell wood; the lumber merchants already do that.

Sounds good. I'll rein it in then.

**Comment by Kywi001** [comment]
Additional on the #59749
The pocket size issue also exists on all trader in the location, such as the nurse and doctor, both won't accept even Merch due to them having 0 pocket size, I had to debug equip them with duffel bags so that they would accept the trade.

**Comment by Merikona** [comment]
> Additional on the #59749 The pocket size issue also exists on all trader in the location, such as the nurse and doctor, both won't accept even Merch due to them having 0 pocket size, I had to debug equip them with duffel bags so that they would accept the trade.

I can take a look at this as well.

**Comment by Merikona** [comment]
I can't seem to make it so the scrapper refuse to sell the containers for metals by themselves. e.g. You can't purchase platinum from him until after you do his quest, but you can buy the bags that hold the platinum. 

I also can't make it so the scrapper would refuse to sell his gun and possessions, so instead I made prices for them that are way over market value. $200 for his AR15, $100 for his tools, and $200 for the cash register...

If anyone knows a way to fix these issues, I'd love to do so before merging this PR.

**Comment by andrei8l** [comment]
> I can't seem to make it so the scrapper refuse to sell the containers for metals by themselves. e.g. You can't purchase platinum from him until after you do his quest, but you can buy the bags that hold the platinum.

Here's an example for solving this:
```diff
diff --git a/data/json/npcs/scrap_trader/scrap_trader.json b/data/json/npcs/scrap_trader/scrap_trader.json
index a0b39e704e..a5d1d00aca 100644
--- a/data/json/npcs/scrap_trader/scrap_trader.json
+++ b/data/json/npcs/scrap_trader/scrap_trader.json
@@ -38,6 +38,12 @@
     "carry_override": "SCRAP_MERCHANT_carried",
     "shopkeeper_item_group": [
       { "group": "SCRAPPER_Shop", "rigid": true },
+      {
+        "group": "SCRAPPER_Bags",
+        "rigid": true,
+        "condition": { "compare_int": [ { "npc_val": "var", "var_name": "NEVER" }, "==", { "const": 1 } ] },
+        "refusal": "just buy the contents pal"
+      },
       {
         "group": "SCRAPPER_Shop_Specialty_Metals",
         "rigid": true,
@@ -144,6 +150,12 @@
       { "item": "platinum_small", "count": [ 0, 3 ], "container-item": "jar_glass_sealed", "charges": 100 }
     ]
   },
+  {
+    "type": "item_group",
+    "id": "SCRAPPER_Bags",
+    "subtype": "collection",
+    "items": [ { "item": "bag_zipper", "count": 0 }, { "item": "jar_glass_sealed", "count": 0 } ]
+  },
   {
     "type": "talk_topic",
     "id": "TALK_NPC_SCRAP_TRADER",
```

**Comment by MNG-cataclysm** [comment]
I also noticed that there is no bed in the corner of the trader's office, while the scrap trader mentions it.

**Comment by Merikona** [comment]
> I also noticed that there is no bed in the corner of the trader's office, while the scrap trader mentions it.

Yea, that's true. I haven't really messed with mapgen before. I could put a couple of beds in the corner, but the space is already very cramped. Let me see if I can shuffle stuff around outside the office to make a shed or something.

**Comment by andrei8l** [comment]
You should add a trade zone over the counter (a `LOOT_UNSORTED` owned by the scrapper's faction). I still haven't added documentation for that, but there's an example [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/mapgen/exodii/exodii_nested.json#L17).

**Comment by Merikona** [comment]
Whew, took me a while to get this done, but I think it's finally ready.

**Comment by andrei8l** [comment]
Instead of blocking the fridge items, just make the loot zones not cover the fridge. One of your loot zones is over a wall and out of range too. You can verify the placements with debug mode (action menu->debug->toggle debug mode, then open the zone editor and change `F`action).

File: data/json/npcs/scrap_trader/scrap_trader.json [hunk 1]
```diff
@@ -2,14 +2,179 @@
...
+      { "trait": "STRONGSTOMACH" },
+      { "trait": "PAINRESIST" },
+      { "trait": "NIGHTVISION" },
+      { "trait": "QUICK" },
+      { "trait": "GOODCARDIO" },
+      { "trait": "TOUGH" },
+      { "trait": "ADRENALINE" },
+      { "trait": "NARCOLEPTIC" },
+      { "trait": "PROJUNK" },
+      { "trait": "MOODSWINGS" }
```
Comment (MNG-cataclysm) [inline data/json/npcs/scrap_trader/scrap_trader.json:35]:
```suggestion
```
Why does the scrapper have mood swings? It seems unnecessary and problematic.

**Review by MNG-cataclysm — CHANGES_REQUESTED** [review]

File: data/json/npcs/scrap_trader/scrap_trader.json (hunk 1 above)
Comment (Merikona) [reply data/json/npcs/scrap_trader/scrap_trader.json:35]: Why is it problematic? It's just there for flavor.

Comment (MNG-cataclysm) [reply data/json/npcs/scrap_trader/scrap_trader.json:35]: In that case, I like it.

**Comment by MNG-cataclysm** [comment]
You should probably mention that this PR will fix issue #61295.

File: src/npc.cpp [hunk 2]
```diff
@@ -2124,9 +2124,6 @@ void npc::shop_restock()
             }
         }
     }
-    if( value_groups.empty() && rigid_groups.empty() ) {
-        return;
-    }
```
Comment (Fris0uman) [inline src/npc.cpp:2127]: What's this change for?

Comment (Merikona) [reply src/npc.cpp:2127]: It's from a comment from andrei. See here. https://discord.com/channels/598523535169945603/598529174302490644/1003540642825257050

**Pull request merged**